src/core/parsers.rs

```diff
@@ -37,19 +21,23 @@ pub fn nix_stat_parser(
 ) -> FileStat {
     let mut atime = String::with_capacity(15);
     let mut mtime = String::with_capacity(15);
-    let ctime = String::new();
+    let mut ctime = String::new();
 
+    // Each line contains one timestamp
+    let mut index = 0;
     for line in stream.lines() {
-        if line.contains("Access") && !line.contains("Uid") {
-            atime = nix_timestamp_parser(line);
-        } else if line.contains("Modify") {
-            mtime = nix_timestamp_parser(line)
-        }
+	if index == 0 {
+            ctime = nix_timestamp_parser(line);
+	} else if index == 1 {
+	    atime = nix_timestamp_parser(line);	    
+	} else if index == 2 {
+	    mtime = nix_timestamp_parser(line);
+	}
+	index = index + 1;
     }
```

Comment:
I think it's more idiomatic to use `match` here instead:

```rust
let mut index = 0;

for line in stream.lines() {
    match index {
        0 => ctime = nix_timestamp_parser(line),
        1 => atime = nix_timestamp_parser(line),
        2 => mtime = nix_timestamp_parser(line)
    }

    index = index + 1;
}
```

Reply:
You're right, it's better with a match. I didn't think about it. It seems I cannot just accept your change from here, I'll integrate them in a commit.